pkg/git/clone.go

```diff
@@ -44,17 +44,30 @@ func (c *Clone) Download(config *api.Config) error {
 			path := filepath.Join(targetSourceDir, config.ContextDir) + string(filepath.Separator) + "."
 			err := c.Copy(path, originalTargetDir)
 			if err != nil {
-				return err
+				return nil, err
 			}
+			info = c.GetInfo(targetSourceDir)
 			c.RemoveDirectory(targetSourceDir)
+		} else {
+			info = c.GetInfo(targetSourceDir)
 		}
 
-		return nil
+		return info, nil
 	}
 	// we want to copy entire dir contents, thus we need to use dir/. construct
 	path := filepath.Join(config.Source, config.ContextDir) + string(filepath.Separator) + "."
 	if !c.Exists(path) {
-		return errors.NewSourcePathError(path)
+		return nil, errors.NewSourcePathError(path)
+	}
+	if err := c.Copy(path, targetSourceDir); err != nil {
```

Comment:
I'd try getting info before doing copy, because if I do (my 3rd test case):
```
sti build 3.3/test/setup-test-app/ openshift/python-33-centos7 python-sample-app
```
From inside sti-python this could return valid info, assuming you do that in the source (represented by `path` here), but after copying all the git info is lost.